<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Setting up a project &mdash; multiome-wf  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="How to get started" href="tutorial.html" />
    <link rel="prev" title="About" href="../index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            multiome-wf
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Setting up a project</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#installation">1. Installation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#conda">Conda</a></li>
<li class="toctree-l3"><a class="reference internal" href="#clone-repository">Clone Repository</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#configure">2. Configure</a></li>
<li class="toctree-l2"><a class="reference internal" href="#run">3. Run</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tutorial.html">How to get started</a></li>
<li class="toctree-l1"><a class="reference internal" href="why-use.html">Why use <cite>multiome-wf</cite>?</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Workflows</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-wf.html">Overview of workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/multiome.html">Multiome Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/rna.html">scRNA-Seq Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/atac.html">scATAC-Seq Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-input.html">Overview of input files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-output.html">Overview of output files</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Configuration</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../configuration/config.html">Configuration details</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/config-yaml.html">Config YAML</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/samples-table.html">Samples Table</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/aggregates-table.html">Aggregates Table</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/assays-table.html">Assays Table</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/config-multiome.html">Multiome Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/config-rna.html">scRNA-Seq Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../configuration/config-atac.html">scATAC-Seq Examples</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Project Info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../project-info/citations.html">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/authors.html">Development Team</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">multiome-wf</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Setting up a project</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/getting-started/setup.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="setting-up-a-project">
<span id="setup"></span><h1>Setting up a project<a class="headerlink" href="#setting-up-a-project" title="Link to this heading"></a></h1>
<p>The general steps to use <cite>multiome-wf</cite> in a new project are:</p>
<ol class="arabic simple">
<li><p><strong>Installation:</strong> download the repository to your local system, into the place
where you want to perform the data analysis. Run the following command in your
terminal:</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Option 1: using SSH keys</span>
$<span class="w"> </span>git<span class="w"> </span>clone<span class="w"> </span>git@github.com:NICHD-BSPC/multiome-wf.git<span class="w"> </span>&lt;project_name&gt;

<span class="c1"># Option 2: using HTTPS</span>
$<span class="w"> </span>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/NICHD-BSPC/multiome-wf.git<span class="w"> </span>&lt;project_name&gt;
</pre></div>
</div>
<p>2. <strong>Configure:</strong> set up <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code> table for experiments and edit configuration file.
Optionally, set up <code class="docutils literal notranslate"><span class="pre">aggregates.tsv</span></code> for cellranger-aggr barcode/library mapping
and <code class="docutils literal notranslate"><span class="pre">assays.tsv</span></code> if you calculated custom count matrices.</p>
<ol class="arabic simple" start="3">
<li><p><strong>Run:</strong> activate environment and run the Snakemake file either locally or on a cluster</p></li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p><cite>multiome-wf</cite> is tested and heavily used on Linux.</p></li>
<li><p>It is likely to work on macOS as long as all relevant conda packages are
available for macOS – though this is not tested.</p></li>
<li><p>It will <strong>not</strong> work on Windows due to a general lack of support of Windows
in bioinformatics tools.</p></li>
<li><p>The workflow is optimized based on the software versions in the <code class="docutils literal notranslate"><span class="pre">env.yaml</span></code>.</p></li>
<li><p>Snakemake versions compatible with the current verion of <cite>multiome-wf</cite> are
<code class="docutils literal notranslate"><span class="pre">&gt;=7.29</span></code> and <code class="docutils literal notranslate"><span class="pre">&lt;8.0</span></code></p></li>
<li><p>In the current version of <cite>multiome-wf</cite>, input datasets mapped to the <strong>mm10
(mouse)</strong> or <strong>hg38 (human)</strong> reference genomes are compatible.</p></li>
</ul>
</div>
<section id="installation">
<h2>1. Installation<a class="headerlink" href="#installation" title="Link to this heading"></a></h2>
<section id="conda">
<h3>Conda<a class="headerlink" href="#conda" title="Link to this heading"></a></h3>
<p>The main prerequisite for <cite>multiome-wf</cite> is <a class="reference external" href="https://docs.conda.io/en/latest/">conda</a>,
with the <a class="reference external" href="https://bioconda.github.io">bioconda</a> channel set up and
the <a class="reference external" href="https://github.com/mamba-org/mamba">mamba</a> drop-in replacement for conda.</p>
<p>If you have not done so already, install conda or miniconda and then mamba into your
base environment. See links above for details.</p>
</section>
<section id="clone-repository">
<h3>Clone Repository<a class="headerlink" href="#clone-repository" title="Link to this heading"></a></h3>
<p>Clone this repository into a project directory, using the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/NICHD-BSPC/multiome-wf.git
</pre></div>
</div>
<p>If you have SSH keys set up for GitHub, run the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>git<span class="w"> </span>clone<span class="w"> </span>git@github.com:NICHD-BSPC/multiome-wf.git
</pre></div>
</div>
<p>Enter the project directory and create an environment based on an environment
<a class="reference external" href="https://en.wikipedia.org/wiki/YAML">YAML</a> file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>multiome-wf
$<span class="w"> </span>mamba<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>--prefix<span class="w"> </span>./env<span class="w"> </span>--file<span class="w"> </span>env.yaml
</pre></div>
</div>
<p>Since we specify a YAML environment file, be sure to use <code class="docutils literal notranslate"><span class="pre">mamba</span> <span class="pre">env</span> <span class="pre">create</span></code>
instead of the more common <code class="docutils literal notranslate"><span class="pre">mambe</span> <span class="pre">create</span></code> command.</p>
</section>
</section>
<section id="configure">
<h2>2. Configure<a class="headerlink" href="#configure" title="Link to this heading"></a></h2>
<p>This step takes the most effort. The first time you set up a project it
will take some time to understand the configuration system.</p>
<p>See <a class="reference internal" href="../configuration/config.html#config"><span class="std std-ref">Configuration details</span></a> for more.</p>
</section>
<section id="run">
<h2>3. Run<a class="headerlink" href="#run" title="Link to this heading"></a></h2>
<p>Activate the main conda environment and go to the workflow you want to run.
For example if you have configured a scRNA-seq run, then do:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>conda<span class="w"> </span>activate<span class="w"> </span>./env
</pre></div>
</div>
<p>and run the following:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Dry run</span>
$<span class="w"> </span>snakemake<span class="w"> </span>-n
</pre></div>
</div>
<p>If all goes well, this should print a list of jobs to be run.</p>
<p>You can run locally, but this is NOT recommended. To run locally, choose the
number of CPUs you want to use with the <code class="docutils literal notranslate"><span class="pre">-j</span></code> argument as is standard for
Snakemake.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If you haven’t made any changes to the
<a class="reference external" href="https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html">Snakefile</a>,
be aware that the default configuration needs a lot of RAM. Adjust the Snakefiles
accordingly if you don’t have enough RAM available (search for “Xmx” to find
the Java args that set memory).</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># run locally (not recommended)</span>
snakemake<span class="w"> </span>--use-conda<span class="w"> </span>-j<span class="w"> </span><span class="m">8</span>
</pre></div>
</div>
<p>and then monitor the various jobs that will be submitted on your behalf. See
<a class="reference internal" href="../configuration/config.html#cluster"><span class="std std-ref">Running on a cluster</span></a> for more details on this.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can execute Snakemake jobs on a cluster using
<a class="reference external" href="https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles">cluster profiles</a>.
Consult the configuration of your high-performance computing system. The current pipeline
relies on the <a class="reference external" href="https://github.com/NIH-HPC/snakemake_profile">snakemake_profile</a>
supported by <a class="reference external" href="https://hpc.nih.gov/">NIH Biowulf</a>.</p>
</div>
<p>You can typically run simultaneous workflows when they are in different
directories; see <a class="reference internal" href="../workflows/overview-wf.html#overview-wf"><span class="std std-ref">Overview of workflows</span></a> for details.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../index.html" class="btn btn-neutral float-left" title="About" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="tutorial.html" class="btn btn-neutral float-right" title="How to get started" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Mira Sohn, Christopher Rhodes.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>