<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Samples Table &mdash; multiome-wf  documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=19f00094" />

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=5929fcd5"></script>
        <script src="../_static/doctools.js?v=9a2dae69"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Aggregates Table" href="aggregates-table.html" />
    <link rel="prev" title="Config YAML" href="config-yaml.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            multiome-wf
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting-started/setup.html">Setting up a project</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting-started/tutorial.html">How to get started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting-started/why-use.html">Why use <cite>multiome-wf</cite>?</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Workflows</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-wf.html">Overview of workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/multiome.html">Multiome Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/rna.html">scRNA-Seq Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/atac.html">scATAC-Seq Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-input.html">Overview of input files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../workflows/overview-output.html">Overview of output files</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Configuration</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="config.html">Configuration details</a></li>
<li class="toctree-l1"><a class="reference internal" href="config-yaml.html">Config YAML</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Samples Table</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#field-descriptions">Field descriptions</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sample"><code class="docutils literal notranslate"><span class="pre">sample</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#replicate"><code class="docutils literal notranslate"><span class="pre">replicate</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#genome"><code class="docutils literal notranslate"><span class="pre">genome</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#hdf5-multiple-assays"><code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#rds-multiple-assays"><code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#gene-expression"><code class="docutils literal notranslate"><span class="pre">Gene.Expression</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#peaks"><code class="docutils literal notranslate"><span class="pre">Peaks</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tf"><code class="docutils literal notranslate"><span class="pre">TF</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#fragments"><code class="docutils literal notranslate"><span class="pre">fragments</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#singlecell"><code class="docutils literal notranslate"><span class="pre">singlecell</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#meta"><code class="docutils literal notranslate"><span class="pre">meta_*</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#examples">Examples</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#multiome">Multiome</a></li>
<li class="toctree-l3"><a class="reference internal" href="#rna-seq">RNA-seq</a></li>
<li class="toctree-l3"><a class="reference internal" href="#atac-seq">ATAC-seq</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="aggregates-table.html">Aggregates Table</a></li>
<li class="toctree-l1"><a class="reference internal" href="assays-table.html">Assays Table</a></li>
<li class="toctree-l1"><a class="reference internal" href="config-multiome.html">Multiome Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="config-rna.html">scRNA-Seq Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="config-atac.html">scATAC-Seq Examples</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Project Info</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../project-info/citations.html">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../project-info/authors.html">Development Team</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">multiome-wf</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Samples Table</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/configuration/samples-table.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="samples-table">
<span id="id1"></span><h1>Samples Table<a class="headerlink" href="#samples-table" title="Link to this heading"></a></h1>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Used for Multiome</p></th>
<th class="head"><p>Used for RNA-seq</p></th>
<th class="head"><p>Used for ATAC-seq</p></th>
<th class="head"><p>Required</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-sample"><span class="std std-ref">sample</span></a></p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-replicate"><span class="std std-ref">replicate</span></a></p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-genome"><span class="std std-ref">genome</span></a></p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>yes</p></td>
<td><p>yes, ATAC/Multiome</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-hdf5"><span class="std std-ref">HDF5_Multiple_Assay</span></a></p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>See notes below</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-rds"><span class="std std-ref">RDS_Multiple_Assays</span></a></p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>See notes below</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-rna"><span class="std std-ref">Gene.Expression</span></a></p></td>
<td><p>yes</p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>See notes below</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-peaks"><span class="std std-ref">Peaks</span></a></p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>yes</p></td>
<td><p>See notes below</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-tf"><span class="std std-ref">TF</span></a></p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>yes</p></td>
<td><p>See notes below</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-frags"><span class="std std-ref">fragments</span></a></p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>yes</p></td>
<td><p>yes, ATAC/Multiome</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-singlecell"><span class="std std-ref">singlecell</span></a></p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
<td><p>yes</p></td>
<td><p>no</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#samples-metadata"><span class="std std-ref">meta_batch</span></a></p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#samples-metadata"><span class="std std-ref">meta_geno</span></a></p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
<td><p>optional</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>Suppored input file formats include HDF5 (10X Genomics compatible),
MEX directory, or RDS.</p></li>
<li><p>For input from non-10X Genomics pipelines, place the <code class="docutils literal notranslate"><span class="pre">barcodes.tsv.gz</span></code>,
<code class="docutils literal notranslate"><span class="pre">features.tsv.gz</span></code>, and <code class="docutils literal notranslate"><span class="pre">matrix.mtx.gz</span></code> in the MEX directory.</p></li>
<li><p>For details about 10X Genomics’ HDF5 format, refer to the <a class="reference external" href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/h5_matrices">HDF5 Feature-Barcode Matrix Format</a>.</p></li>
</ul>
</div>
<section id="field-descriptions">
<h2>Field descriptions<a class="headerlink" href="#field-descriptions" title="Link to this heading"></a></h2>
<section id="sample">
<span id="samples-sample"></span><h3><code class="docutils literal notranslate"><span class="pre">sample</span></code><a class="headerlink" href="#sample" title="Link to this heading"></a></h3>
<blockquote>
<div><p>string. Defines labels for each sample. Values in <code class="docutils literal notranslate"><span class="pre">sample</span></code> column
must be unique, unless analyzing technical replicates. If specifying
technical replicates, sample label must be the same for all rows
containing techical replicates.</p>
</div></blockquote>
</section>
<section id="replicate">
<span id="samples-replicate"></span><h3><code class="docutils literal notranslate"><span class="pre">replicate</span></code><a class="headerlink" href="#replicate" title="Link to this heading"></a></h3>
<blockquote>
<div><p>string. Optional. Defines labels for each technical replicate of
a sample. Values in <code class="docutils literal notranslate"><span class="pre">replicate</span></code> column must be unique per sample.
<strong>If not specified</strong> technical replicates, leave empty. <strong>If specified</strong>
technical replicates: sample must be the same for all rows containing
techical replicates.</p>
</div></blockquote>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Refer to the <a class="reference internal" href="../workflows/overview-input.html#replicates"><span class="std std-ref">Replicates</span></a> section for the definition of biological and
technical replicates.</p>
</div>
</section>
<section id="genome">
<span id="samples-genome"></span><h3><code class="docutils literal notranslate"><span class="pre">genome</span></code><a class="headerlink" href="#genome" title="Link to this heading"></a></h3>
<blockquote>
<div><p>string, default <code class="docutils literal notranslate"><span class="pre">hg38</span></code>. Defines labels for genome build per sample.
If samples include “Peaks” or “Transcription Factor” (TF) matrices,
as with ATAC and Multiome products, all samples must have genome.
Otherwise, genome labels can differ across samples. In the current
version of the <cite>multiome-wf</cite>, <code class="docutils literal notranslate"><span class="pre">mm10</span></code> (mouse) and <code class="docutils literal notranslate"><span class="pre">hg38</span></code> (human) are
available.</p>
</div></blockquote>
</section>
<section id="hdf5-multiple-assays">
<span id="samples-hdf5"></span><h3><code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code><a class="headerlink" href="#hdf5-multiple-assays" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to HDF5 file containing <strong>multiple</strong> feature-by-barcode matrices.
If a sample’s input contains multiple matrices, list the sample path
in either <code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code> or <code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code> columns
of <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>. Paths specified in <code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code> should
point to a 10X Genomics compatible HDF5 file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>If the specified input file contains feature-by-barcode matrices f
or 5’ or 3’ Gene Expression counts matrix, or ATAC/Multiome counts
matrices including Peaks or Transcription Factors, <strong>do not</strong> duplicate
paths in <code class="docutils literal notranslate"><span class="pre">Gene.Expression</span></code>, <code class="docutils literal notranslate"><span class="pre">Peaks</span></code> or <code class="docutils literal notranslate"><span class="pre">TF</span></code> columns, respectively.</p></li>
<li><p>Specifying paths in this column is useful if you are analyzing 10X Genomics
kits containing multiple assays, such as Gene Expression + CRISPR
barcodes (Perturb-Seq), 3’ Gene Expression + Protein barcodes (CITE-Seq),
Multiome (Gene Expression + ATAC), etc.</p></li>
<li><p>Specifying paths in this column can also be useful if you are analyzing
data derived from non-10X Genomics platforms, which may need reformatting
prior to running the workflow.</p></li>
</ul>
</div>
</div></blockquote>
</section>
<section id="rds-multiple-assays">
<span id="samples-rds"></span><h3><code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code><a class="headerlink" href="#rds-multiple-assays" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to RDS file containing <strong>multiple</strong> feature-by-barcode matrices saved
in a named list of dgCMatrices. The saved data format is identical to that of
<a class="reference internal" href="#samples-hdf5"><span class="std std-ref">HDF5_Multiple_Assays</span></a> (Refer to the notes under <a class="reference internal" href="#samples-hdf5"><span class="std std-ref">HDF5_Multiple_Assays</span></a>).</p>
</div></blockquote>
</section>
<section id="gene-expression">
<span id="samples-rna"></span><h3><code class="docutils literal notranslate"><span class="pre">Gene.Expression</span></code><a class="headerlink" href="#gene-expression" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to a <strong>single</strong> feature-by-barcode matrix containing 5’ or 3’ Gene
Expression counts. Path can point to either 1) a file in HDF or RDS format,
or 2) the parent directory containing a 10X compatible MEX sparse matrix and
associated barcodes and features files.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>Specifying paths in this column is useful if you are analyzing 10X
Genomics single-assay transcriptomics kits.</p></li>
<li><p>If analyzing single cell transcriptomics data from a non-10X Genomics
platform, reformatting the gene-by-barcode count matrix is likely
needed. In such a case, the reformatted counts matrix can be save in HDF5
or RDS format and specified in the <code class="docutils literal notranslate"><span class="pre">Gene.Expression</span></code> column of
<code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>.</p></li>
<li><p>If the specified matrix has any additional features, for example 3’ Gene
Expression + CRISPR barcodes (Perturb-Seq), 3’ Gene Expression + Protein
barcodes (CITE-Seq), use either <code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code> or
<code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code> columns of <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>.</p></li>
</ul>
</div>
</div></blockquote>
</section>
<section id="peaks">
<span id="samples-peaks"></span><h3><code class="docutils literal notranslate"><span class="pre">Peaks</span></code><a class="headerlink" href="#peaks" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to a <strong>single</strong> feature-by-barcode matrix containing ATAC peaks counts.
Path can point to either 1) a file in HDF or RDS format, or 2) the parent
directory containing a 10X compatible MEX sparse matrix and associated barcodes
and features files.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>Specifying paths in this column is useful if you are analyzing 10X
Genomics single-assay chromatin accessibility kits.</p></li>
<li><p>If analyzing single cell ATAC data from a non-10X platform,
reformatting the gene-by-barcode count matrix is likely needed. In such
a case, the reformatted counts matrix can be save in HDF5 or RDS format
and specified in the <code class="docutils literal notranslate"><span class="pre">Peaks</span></code> column of <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>.</p></li>
<li><p>If the specified matrix has any additional features, for example Multiome
(Gene Expression + ATAC), use either <code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code> or
<code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code> columns of <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>.</p></li>
</ul>
</div>
</div></blockquote>
</section>
<section id="tf">
<span id="samples-tf"></span><h3><code class="docutils literal notranslate"><span class="pre">TF</span></code><a class="headerlink" href="#tf" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to a <strong>single</strong> feature-by-barcode matrix encoding Transcription Factor
counts. Path can point to either 1) a file in HDF or RDS format, or 2) the
parent directory containing a 10X compatible MEX sparse matrix and associated
barcodes and features files. Refer to the notes under <a class="reference internal" href="#samples-peaks"><span class="std std-ref">Peaks</span></a>.</p>
</div></blockquote>
</section>
<section id="fragments">
<span id="samples-frags"></span><h3><code class="docutils literal notranslate"><span class="pre">fragments</span></code><a class="headerlink" href="#fragments" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to the <code class="docutils literal notranslate"><span class="pre">fragments.tsv.gz</span></code> file containing a table of tagmentation loci,
each with coordinates and de-duplicated counts, created from 10X Genomics ATAC
and Multiome kits.</p>
</div></blockquote>
</section>
<section id="singlecell">
<span id="samples-singlecell"></span><h3><code class="docutils literal notranslate"><span class="pre">singlecell</span></code><a class="headerlink" href="#singlecell" title="Link to this heading"></a></h3>
<blockquote>
<div><p>Path to the <code class="docutils literal notranslate"><span class="pre">singlecell.csv</span></code> file created from 10X Genomics ATAC and Multiome
kits.</p>
</div></blockquote>
</section>
<section id="meta">
<span id="samples-metadata"></span><h3><code class="docutils literal notranslate"><span class="pre">meta_*</span></code><a class="headerlink" href="#meta" title="Link to this heading"></a></h3>
<blockquote>
<div><p>string. Optional. Define columns for metadata labels. Columns beginning
with <code class="docutils literal notranslate"><span class="pre">meta_</span></code> are placeholders for user-specified metadata. Users can
rename these columns.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul class="simple">
<li><p>In <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code>, the following column names are considered immutable:
<code class="docutils literal notranslate"><span class="pre">sample</span></code>, <code class="docutils literal notranslate"><span class="pre">replicate</span></code>, <code class="docutils literal notranslate"><span class="pre">genome</span></code>, <code class="docutils literal notranslate"><span class="pre">HDF5_Multiple_Assays</span></code>,
<code class="docutils literal notranslate"><span class="pre">RDS_Multiple_Assays</span></code>, <code class="docutils literal notranslate"><span class="pre">Gene.Expression</span></code>, <code class="docutils literal notranslate"><span class="pre">Peaks</span></code>, <code class="docutils literal notranslate"><span class="pre">TF</span></code>,
<code class="docutils literal notranslate"><span class="pre">fragments</span></code>, <code class="docutils literal notranslate"><span class="pre">singlecell</span></code>.</p></li>
<li><p>Any additional columns present in <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code> will be considered
metadata columns. Metadata columns can have any unique label, not just
the <code class="docutils literal notranslate"><span class="pre">meta_*</span></code> used in the example samples table.</p></li>
</ul>
</div>
</div></blockquote>
</section>
</section>
<section id="examples">
<h2>Examples<a class="headerlink" href="#examples" title="Link to this heading"></a></h2>
<section id="multiome">
<h3>Multiome<a class="headerlink" href="#multiome" title="Link to this heading"></a></h3>
<p>A <strong>basic</strong> examples of a <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code> file for 10X Genomics
Multiome analysis is below:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>replicate</p></th>
<th class="head"><p>genome</p></th>
<th class="head"><p>HDF5_Multiple_Assays</p></th>
<th class="head"><p>RDS_Multiple_Assays</p></th>
<th class="head"><p>Gene.Expression</p></th>
<th class="head"><p>Peaks</p></th>
<th class="head"><p>TF</p></th>
<th class="head"><p>fragments</p></th>
<th class="head"><p>singlecell</p></th>
<th class="head"><p>meta_batch</p></th>
<th class="head"><p>meta_geno</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>rep1_wt</p></td>
<td></td>
<td><p>mm10</p></td>
<td><p>rep1_wt/outs/filtered_feature_bc_matrix.h5</p></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><p>rep1_wt/outs/atac_fragments.tsv.gz</p></td>
<td></td>
<td><p>batch1</p></td>
<td><p>wt</p></td>
</tr>
<tr class="row-odd"><td><p>rep1_homo</p></td>
<td></td>
<td><p>mm10</p></td>
<td><p>rep_homo/outs/filtered_feature_bc_matrix.h5</p></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><p>rep1_homo/outs/atac_fragments.tsv.gz</p></td>
<td></td>
<td><p>batch1</p></td>
<td><p>ko</p></td>
</tr>
</tbody>
</table>
</section>
<section id="rna-seq">
<h3>RNA-seq<a class="headerlink" href="#rna-seq" title="Link to this heading"></a></h3>
<p>A <strong>basic</strong> examples of a <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code> file for 10X Genomics
Single Cell 3’ Gene Expression analysis is below:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>replicate</p></th>
<th class="head"><p>genome</p></th>
<th class="head"><p>HDF5_Multiple_Assays</p></th>
<th class="head"><p>RDS_Multiple_Assays</p></th>
<th class="head"><p>Gene.Expression</p></th>
<th class="head"><p>Peaks</p></th>
<th class="head"><p>TF</p></th>
<th class="head"><p>fragments</p></th>
<th class="head"><p>singlecell</p></th>
<th class="head"><p>meta_batch</p></th>
<th class="head"><p>meta_tissue</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CTX</p></td>
<td></td>
<td><p>mm10</p></td>
<td></td>
<td></td>
<td><p>wt/outs/filtered_feature_bc_matrix.h5</p></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><p>batch1</p></td>
<td><p>CTX</p></td>
</tr>
<tr class="row-odd"><td><p>MGE</p></td>
<td></td>
<td><p>mm10</p></td>
<td></td>
<td></td>
<td><p>ko/outs/filtered_feature_bc_matrix.h5</p></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><p>batch1</p></td>
<td><p>MGE</p></td>
</tr>
</tbody>
</table>
</section>
<section id="atac-seq">
<h3>ATAC-seq<a class="headerlink" href="#atac-seq" title="Link to this heading"></a></h3>
<p>A <strong>basic</strong> examples of a <code class="docutils literal notranslate"><span class="pre">samples.tsv</span></code> file for 10X Genomics
Single Cell ATAC analysis is below:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>sample</p></th>
<th class="head"><p>replicate</p></th>
<th class="head"><p>genome</p></th>
<th class="head"><p>HDF5_Multiple_Assays</p></th>
<th class="head"><p>RDS_Multiple_Assays</p></th>
<th class="head"><p>Gene.Expression</p></th>
<th class="head"><p>Peaks</p></th>
<th class="head"><p>TF</p></th>
<th class="head"><p>fragments</p></th>
<th class="head"><p>singlecell</p></th>
<th class="head"><p>meta_batch</p></th>
<th class="head"><p>meta_tissue</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>CTX</p></td>
<td></td>
<td><p>mm10</p></td>
<td></td>
<td></td>
<td></td>
<td><p>CTX_rerun/outs/filtered_peak_bc_matrix.h5</p></td>
<td><p>CTX_rerun/outs/filtered_tf_bc_matrix.h5</p></td>
<td><p>CTX_rerun/outs/fragments.tsv.gz</p></td>
<td><p>CTX_rerun/outs/summary.csv</p></td>
<td><p>batch1</p></td>
<td><p>CTX</p></td>
</tr>
<tr class="row-odd"><td><p>MGE</p></td>
<td></td>
<td><p>mm10</p></td>
<td></td>
<td></td>
<td></td>
<td><p>MGE_rerun/outs/filtered_peak_bc_matrix.h5</p></td>
<td><p>MGE_rerun/outs/filtered_tf_bc_matrix.h5</p></td>
<td><p>MGE_rerun/outs/fragments.tsv.gz</p></td>
<td><p>MGE_rerun/outs/summary.csv</p></td>
<td><p>batch1</p></td>
<td><p>MGE</p></td>
</tr>
</tbody>
</table>
<p>See <a class="reference internal" href="../workflows/overview-wf.html#overview-wf"><span class="std std-ref">Overview of workflows</span></a> for more detailed examples of config files.</p>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="config-yaml.html" class="btn btn-neutral float-left" title="Config YAML" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="aggregates-table.html" class="btn btn-neutral float-right" title="Aggregates Table" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, Mira Sohn, Christopher Rhodes.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>